src/light-stemcell-builder/driver/create_machine_image_driver.go

```diff
@@ -80,15 +79,7 @@ func (d *SDKCreateMachineImageDriver) Create(driverConfig resources.MachineImage
 
 	d.logger.Printf("finished uploaded image to s3 after %f minutes\n", time.Since(uploadStartTime).Minutes())
 
-	getReq, _ := d.s3Client.GetObjectRequest(&s3.GetObjectInput{
-		Bucket: aws.String(driverConfig.BucketName),
-		Key:    aws.String(keyName),
-	})
-
-	machineImageGetURL, err := getReq.Presign(2 * time.Hour)
-	if err != nil {
-		return resources.MachineImage{}, fmt.Errorf("failed to sign GET request: %s", err)
-	}
+	machineImageGetURL := "s3://" + driverConfig.BucketName + "/" + keyName
```

Comment:
I'm a bit confused by this line in the context of this commit. This Driver is only used by the standard region publisher, the [isolated region publisher](https://github.com/cloudfoundry/bosh-aws-light-stemcell-builder/blob/c621f84fa888cf70ef8bc175f8dc844aca76c69b/src/light-stemcell-builder/driverset/isolated_aws_region.go#L31) uses the `MachineImageManifestDriver` which also [presigns urls](https://github.com/cloudfoundry/bosh-aws-light-stemcell-builder/blob/c621f84fa888cf70ef8bc175f8dc844aca76c69b/src/light-stemcell-builder/driver/create_machine_image_manifest_driver.go#L146). I also suspect any issues tied to bad pre-signed get urls will also apply to the delete urls as well, but all of this implies that we either can't use private s3 buckets or we need to pass credentials to the other places like: [here](https://github.com/cloudfoundry/bosh-aws-light-stemcell-builder/blob/c621f84fa888cf70ef8bc175f8dc844aca76c69b/src/light-stemcell-builder/driver/delete_machine_image_driver.go#L63). Was there a test that drove out the need for this change or was this just observed behavior?

Reply:
> I'm a bit confused by this line in the context of this commit. This Driver is only used by the standard region publisher, the isolated region publisher uses the MachineImageManifestDriver which also presigns URLs.

We are using the standard region publisher. Do you think we should use the isolated region publisher instead? The preliminary test on us-iso-east shows that ImportSnapshot works as expected. I am not sure if the ImportVolume API is available in the us-iso-east region.

Worth noting, we haven't tested the changes in the us-iso-east region yet.

> I also suspect any issues tied to bad pre-signed get urls will also apply to the delete urls as well, but all of this implies that we either can't use private s3 buckets or we need to pass credentials to the other places like: here.

The issue I ran into was specific to the ImportSnapshot API. The API doesn't accept pre-signed URLs when IAM instance profiles are used. The delete URL, on the other hand, is used with a Go HTTP client, and that should continue to work as expected.

> Was there a test that drove out the need for this change or was this just observed behavior?

We are running the test suite in https://eagle.ci.cf-app.com/teams/main/pipelines/eagle_product_images/jobs/test_bosh_aws_light_stemcell_builder/builds/3 using IAM instance profile. We made those changes to get the tests passing in our pipeline.

Reply:
I can't find much on the history of the isolated region code but from the name, it sounds like it should match your usecase - but yeah I understand if it doesn't have the ImportVolume API that it may be easier to just use Standard. It looks like theres another gov cloud region specified [here](https://github.com/cloudfoundry/bosh-aws-light-stemcell-builder/blob/master/src/light-stemcell-builder/config/config.go#L24) as isolated but I don't know enough about the regions to say if they're similar or not.